frontend/public/extend/devconsole/shared/utils/__tests__/pipeline-utils.spec.ts

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable no-unused-vars, no-undef */
+import { getPipelineTasks } from '../pipeline-utils';
+import { mockPipelinesJSON } from '../__mocks__/PipelineUtilMocks';
+
+describe('Output for task arranged should have the following ', () => {
+  it('For first pipeline there should be 2 stages of length [0:[1],1:[2]]', () => {
```

Comment:
```suggestion
  it('should transform pipelines', () => {
```
This is a good opportunity for snapshot testing as we expect the transformation to be precise.